protocol-units/settlement/mcr/client/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+use movement_types::BlockCommitment;
+use tokio_stream::Stream;
+
+#[cfg(feature = "stub")]
+pub mod stub;
+
+#[cfg(feature = "stub")]
+pub use stub::*;
+
+type CommitmentStream =
+	std::pin::Pin<Box<dyn Stream<Item = Result<BlockCommitment, anyhow::Error>> + Send>>;
+
+#[async_trait::async_trait]
+pub trait McrSettlementClientOperations {
```

Comment:
With the stream discussion, I took a closer look at the code and I have this remark.
The stub McrSettlementClient use send+sync structure to be thread safe.
It's not imposed by the trait. So if the client is means to be used between thread, we should add send+sync to the trait or remove the Arc/Mutex part of the implementation. 
ie :
```
pub trait McrSettlementClientOperations: Send + Sync {
...
}
```

Reply:
I think it's stylistically better to not make auto-traits pre-requisites; you might never need to pass the client itself between threads.